clients/src/main/java/org/apache/kafka/common/PartitionInfo.java

```diff
@@ -108,4 +108,8 @@ private String formatNodeIds(Node[] nodes) {
         return b.toString();
     }
 
+    @Override
+    public int compareTo(PartitionInfo o) {
+        return partition - o.partition();
```

Comment:
I think topic should also be included in comparison. Even though the current usage can work without considering topic,  given that it is a public API and can be used in a more general sense, it is better to be strict and conform to the specification.